File: backend/app/dto/notification.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class NotificationDTO:
    source: str
    organization: str
    exam_name: str
    title: str

    notification_no: Optional[str] = None
    category: Optional[str] = None

    pdf_url: Optional[str] = None
    apply_url: Optional[str] = None
    official_url: Optional[str] = None

    vacancies: Optional[int] = None

    qualification: Optional[str] = None

    age_limit: Optional[str] = None

    salary: Optional[str] = None

    description: Optional[str] = None

File: frontend_old/src/types/notification.ts
export interface Notification {
  id: number;
  source: string;
  organization: string;
  exam_name: string;
  notification_no: string;
  title: string;
  notification_hash: string;
  category: string;
  status: string;
  vacancies: number | null;
  qualification: string | null;
  age_limit: string | null;
  salary: string | null;
  application_start: string | null;
  application_end: string | null;
  exam_date: string | null;
  pdf_url: string | null;
  apply_url: string | null;
  official_url: string | null;
  description: string | null;
  ai_summary: string | null;
  created_at: string;
  updated_at: string;
}